app/controllers/pieces_controller.rb

```diff
@@ -1,16 +1,22 @@
 class PiecesController < ApplicationController
 
-  def show
-    @piece = Piece.find(params[:id])
-    @piece.game.pieces
-  end
-
-
+  def update
+    @game = current_piece.game
 
+    x = params[:x_position]
+    y = params[:y_position]
+    current_piece.update_attributes(:x_position => x, :y_position => y)
```

Comment:
replace `(:x_position => x, :y_position => y)` with `piece_params` and then you can delete:

```
x = params[:x_position]
y = params[:y_position]
```